ludwig/train.py

```diff
@@ -396,6 +400,68 @@ def full_train(
     )
 
 
+def kfold_cross_validate(
+                model_definition,
+                model_definition_file=None,
+                data_csv=None,
+                data_train_csv=None,
+                data_validation_csv=None,
+                output_directory='results',
+                k_fold=None,
+                **kwargs
+):
+    logger = logging.getLogger('ludwig')
+    logger.info('starting {:d}-fold cross validation'.format(k_fold))
+
+    # create output_directory if not available
+    if not os.path.isdir(output_directory):
+        os.mkdir(output_directory)
+
+    # read in data to split for the folds
+    data_df = pd.read_csv(data_train_csv)
+
+    # place each fold in a separate directory
+    data_dir = os.path.dirname(data_train_csv)
+    kfold_training_stats = {}
+    for train_index, test_index, fold_num in generate_kfold_splits(data_train_csv, k_fold):
+        with tempfile.TemporaryDirectory(dir=data_dir) as temp_dir_name:
+            # save training and validation subset for the fold into a temporary directory
+            train_csv_fp = os.path.join(temp_dir_name, 'train_fold.csv')
+            test_csv_fp = os.path.join(temp_dir_name, 'test_fold.csv')
+            logger.info("\n\n>>>>> for fold {:d} created temporary directory: {}".format(fold_num, temp_dir_name))
+            data_df.iloc[train_index].to_csv(train_csv_fp, index=False)
+            data_df.iloc[test_index].to_csv(test_csv_fp, index=False)
+
+            # train and validate model on this fold
+            logger.info("training on fold {:d}".format(fold_num))
+            (model,
+             preprocessed_data,
+             _,
+             train_stats,
+             model_definition) = full_train({},
+                                            model_definition_file=model_definition_file,
+                                            data_train_csv = train_csv_fp,
```

Comment:
in this nad next line please remove the spces before and after the `=` (in general, if you run the pycharm linter with a limit of 80 characters per line, that would be great)

Reply:
completed changes